bin/brew

```diff
@@ -45,5 +85,77 @@ fi
 
 export HOMEBREW_BREW_FILE
 export HOMEBREW_RUBY_PATH
+export HOMEBREW_PREFIX
+export HOMEBREW_REPOSITORY
+export HOMEBREW_LIBRARY
+export HOMEBREW_CELLAR
+
+if [ -n "$HOMEBREW_OSX" ]
+then
+  if [ "$(xcode-\select --print-path)" = "/" ]
+  then
+    cat >&2 <<EOS
+Your xcode-select path is currently set to '/'.
+This causes the 'xcrun' tool to hang, and can render Homebrew unusable.
+If you are using Xcode, you should:
+  sudo xcode-select -switch /Applications/Xcode.app
+Otherwise, you should:
+  sudo rm -rf /usr/share/xcode-select
+EOS
+    exit 1
+  fi
+
+  XCRUN_OUTPUT="$(/usr/bin/xcrun clang 2>&1)"
+  XCRUN_STATUS="$?"
+
+  if [ "$XCRUN_STATUS" != "0" ] && [[ "$XCRUN_OUTPUT" =~ license ]]
+  then
+    cat >&2 <<EOS
+You have not agreed to the Xcode license. Please resolve this by running:
+  sudo xcodebuild -license
+EOS
+    exit 1
+  fi
+fi
+
+# Many Pathname operations use getwd when they shouldn't, and then throw
+# odd exceptions. Reduce our support burden by showing a user-friendly error.
+if ! [ -d "$(pwd)" ]
+then
+  echo "The current working directory doesn't exist, cannot proceed."
+  exit 1
+fi
+
+for i in "$@"
+do
+  [[ "$i" =~ ^- ]] && continue
```

Comment:
Somewhat true. But I think it has a wider impact because it changes the rules how the command is located inside the arguments, which already happens before it's clear which command to invoke and whether that one will be implemented in shell or in Ruby.